spacy/pipeline/pipes.pyx

```diff
@@ -1228,13 +1234,30 @@ class EntityLinker(Pipe):
                 # the gold annotations should link to proper entities - if this fails, the dataset is likely corrupt
                 if not (start, end) in ents_by_offset:
                     raise RuntimeError(Errors.E188)
+
                 ent = ents_by_offset[(start, end)]
 
                 for kb_id, value in kb_dict.items():
                     # Currently only training on the positive instances
                     if value:
                         try:
-                            sentence_docs.append(ent.sent.as_doc())
+                            # find the sentence in the list of sentences.
+                            sent_index = sentences.index(ent.sent)
```

Comment:
The try/except block here should be changed so that it only covers the `ent.sent` part, like this:

```
try: 
    sent = ent.sent
except AttributeError:
    ...
```

We want try/except blocks to be scoped as narrowly as possible, usually over only one statement, because we want the exception to apply to a specific error. If we have a whole block under the `try`, there are lots of places that an `AttributeError` could be raised, so we don't know whether our error message is correct.

Reply:
Yea, point taken @honnibal, my bad. 
@theudas, could you make those changes in this PR as well? Then we can kill two birds with one stone ;-)